client/src/Components/Pagination/Pagination.js

```diff
@@ -0,0 +1,44 @@
+import React, { useState } from 'react';
+import { Pagination } from 'react-bootstrap';
+
+// helper function for generating a range (1,3,1) -> [1,2,3]
+const range = (start, end) => new Array(end - start + 1).fill().map((el, ind) => ind + start);
+
+const PaginationBar = ({ currentPage, setPage, totalPosts, perPage }) => {
+	const totalPages = Math.ceil(totalPosts / perPage);
+	const [shownPages, setShownPages] = useState(range(1, Math.min(totalPages, 3)));
+	function correctShownPages (value) {
+		console.log(totalPosts);
+		if (value === 1) {
+			setShownPages(range(value, Math.min(totalPages, value + 2)));
+		} else if (value === totalPages) {
+			setShownPages(range(Math.max(1, value - 2), value));
+		} else {
+			setShownPages(range(value - 1, value + 1));
+		}
+		console.log(shownPages);
+	}
+
+	function proccessNewPage (page) {
+		return () => {
+			page = Math.max(1, page);
+			page = Math.min(totalPages, page);
+			correctShownPages(page);
+			setPage(page);
+		};
+	}
```

Comment:
This part of the code might look like:
```suggestion
	const proccessNewPage = (page) => () => {
		page = Math.max(1, page);
		page = Math.min(totalPages, page);
		correctShownPages(page);
		setPage(page);
	};
```